### PR TITLE
Feature/models

### PR DESCRIPTION
**Pull request by picksitquick**
## Proposed Changes
Requesting @aprajshekhar to review the changes.
- added new models using the DB tables form gamedoora-data/gamedoora-beta-structure.sql.
- these models are subjected to certain changes as we move forward
- the task to make it's DTOs is underway, will be completed within this week itself.
- We can setup calls to discuss on how to finalize these models, atleast for tasks assigned to students.
- Please take your time in reviewing them as we don't want to miss out on any details.
- Thank you, please provide any feedback necessary.

### CODE REVIEW

File: src/main/java/com/gamedoora/model/db/SchemaMigration.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.gamedoora.model.db;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/db/SchemaMigration.java:1]: This model can be deleted.

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
Overall LGTM. Just a few observations.

File: src/main/java/com/gamedoora/model/db/Identities.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.gamedoora.model.db;
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/model/db/Identities.java:1]: Identities is not required

**Review by aprajshekhar — CHANGES_REQUESTED** [review]

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
Thanks for the PR. I have added some observations.
Two more suggestion for mappers 
- Explicitly specify `componentModel = "spring"`  for ```@Mapper` annotation`
- Add `@Builder`  for all the POJOs.
Once these are completed, this PR can be merged.

**Review by aprajshekhar — APPROVED** [review]
Thanks for the changes. LGTM. Approved.

**Comment by picksitquick** [comment]
I request to pause the merge, @Builder annotation is still creating conflicts, working on it, will be fixed by tomorrow.
Every other changes have been made. creating a repo feature/Error which contain the error code.

**Comment by aprajshekhar** [comment]
> I request to pause the merge, @Builder annotation is still creating conflicts, working on it, will be fixed by tomorrow.
> Every other changes have been made. creating a repo feature/Error which contain the error code.
> 

If builder is creating problems, let's park it for the time being. We can come back to it later. Just create an issue for tracking purposes. Thanks!

**Comment by picksitquick** [comment]
@aprajshekhar 
My apologies for the late reply but, everything now seems to be working, I do need to make unit test cases for mappers which I will update asap, for now, dao, dto and mappers with their implementations are working. Although I would request you to review the changes one last time.

Updates:
1. @Data has been replaced with @Getter and @Setter, using @Data and ignoring default functions like toString() was becoming time consuming, will keep on looking for that but for now this works fine.
2. @Builder has been added to every POJO and is now functional.
3. @Builder requires public constructor while creating objects, which was being missed by @Data annotation, hence an additional @AllArgsConstructor has also been added
4. Lastly, the root cause, spring-boot-starter was not able to process some dependency after adding all these annotation along with map_struct so had to go back to the basics, added spring-context in pom.xml which is ideally the basic spring module, resolved this error.

Not sure whether all of this combined makes it a better approach or not, please let me know if any changes are needed.
This branch can now be merged.

**Comment by aprajshekhar** [comment]
Good job. Before I merge, can you confirm that the model works well with user services?

**Pull request merged**